src/transformers/integrations/fsdp.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from ..utils import is_torch_available
+
+
+if is_torch_available():
+    import torch.distributed.fsdp
+
+if TYPE_CHECKING:
+    from torch import nn
+
+
+def is_fsdp_managed_module(module: nn.Module) -> bool:
+    return isinstance(module, torch.distributed.fsdp.FullyShardedDataParallel) or getattr(
+        module, "_is_fsdp_managed_module", False
+    )
```

Comment:
Can't we use `is_fsdp_enabled` just like how we do it for deepspeed ? I guess users will use fsdp in trainer + accelerate, so that should work. But I understand that it won't work with your general script. cc @muellerzr

Reply:
> But I understand that it won't work with your general script

Right, since `is_fsdp_enabled` is a `Trainer` attribute, it would only work if you are using a `Trainer`, whereas this function works in all scenarios, including when using a `Trainer`.